base/scheduler/CDLL.jl

```diff
@@ -0,0 +1,266 @@
+
+module ConcurrentList #Concurrent Doubly Linked List
+
+mutable struct Node{T}
+    const value::Union{T, Nothing}
+    @atomic next::Union{Node{T}, Nothing}
```

Comment:
You can keep this type stable by having `next` point to itself in case there is no next object.

Reply:
That'd just replace the small union `get_next(node) === nothing` with a `get_next(node) === node` check everywhere though. Not sure it'd help performance, and it'd likely cause some annoying footguns.

Reply:
Intent could at least be made clearer with a helper function `hasnext(node) = get_next(node) != node`.